### PR TITLE
Improve error message when file not found

### PR DESCRIPTION
**Pull request by phillord**
Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):
- [x] The commits are consistent with our [contribution guidelines](../CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [ ] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

The project type is now displayed when a file a matching test or
implementation is not found, which is useful when the project type
has been mis-identified.

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -1946,14 +1946,17 @@ It assumes the test/ folder is at the same level as src/."
       (let ((impl-file (projectile-find-matching-file file-name)))
         (if impl-file
             (projectile-expand-root impl-file)
-          (error "No matching source file found")))
+          (error
+           (format "No matching source file found for project type: %s"
+                   (projectile-project-type)))))
```
Comment (bbatsov) [inline projectile.el]: Or we can simply display the project type at the modeline, which seems like a better idea to me.

**Comment by phillord** [comment]
Nothing wrong with doing both. The mode-line is fine, but mine is overloaded and so I suspect is everyone elses. I don't know how many people diminish projectile, but I bet it's quite a few.

**Comment by yveszoundi** [comment]
Like @phillord, I also doubt that most people diminish projectile. In my configs, I do use `diminish` or similar approaches to avoid cluttering the mode-line, for pretty much everything **but** projectile.

**Comment by bbatsov** [comment]
OK, let's do both things then.

**Comment by bbatsov** [comment]
Ping :-)

**Pull request merged**